trino-aws-proxy/src/main/java/io/trino/aws/proxy/server/credentials/http/HttpCredentialsProvider.java

```diff
@@ -48,13 +60,29 @@ public HttpCredentialsProvider(@ForHttpCredentialsProvider HttpClient httpClient
         this.jsonCodec = requireNonNull(jsonCodec, "jsonCodec is null");
         this.httpCredentialsProviderEndpoint = config.getEndpoint();
         this.httpHeaders = ImmutableMap.copyOf(config.getHttpHeaders());
+        this.credentialsCache = Caffeine.newBuilder()
```

Comment:
If cacheTtl is zero, we wouldn't build or use the cache at all. i.e. it should be `Optional`.

Reply:
I was trying to use a zero-sized cache as a no-op, since it handles the cases where either the TTL or the max size are zero (either of which should result in no caching) nicely.

I agree it's perhaps a bit hacky, will change